lib/elixir/lib/kernel.ex

```diff
@@ -4449,9 +4449,11 @@ defmodule Kernel do
     end
   end
 
-  # TODO: Deprecate by v1.5
   @doc false
+  # TODO: Remove by 2.0
   defmacro to_char_list(arg) do
+    %{file: file, line: line} = __CALLER__
+    :elixir_errors.warn(line, file, "to_char_list/1 is deprecated, use to_charlist/1")
```

Comment:
Is it ok to do this here (since it's at compile time anyways) instead of `elixir_dispatch`? The latter would cause the warning to look like "Kernel.to_char_list/1 ..." instead of "to_char_list/1 ...", maybe that's preferable? I'm not sure.